src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -10,7 +10,13 @@ namespace System.Globalization
 {
     public partial class TextInfo
     {
-        private bool m_needsTurkishCasing;
+        enum TurkishCasing
```

Comment:
Sounds good to me. I have actually noticed that there is the same thread safety bug as discussed in the PR around initialization of `m_IsAsciiCasingSameAsInvariant`. Opened #6794. Perhaps we should make this a more general purpose enum that can be used for both.